tests/pylint_cycles.sh

```diff
@@ -0,0 +1,207 @@
+function join_array_by {
+    local IFS="$1"
+    shift
+    echo "$*"
+}
+
+# generating a .pylintrc to ignore specific circumstances
+# see here for specification: https://github.com/PyCQA/pylint/blob/master/pylintrc
+function create_pylintrc() {
+    local pylintrc=${1}
+    local disabled_pylint_warning
+    disabled_pylint_warning=$(join_array_by ',' "${IGNORED_PYLINT_ERRORS[@]}")
+
+    echo "Generating ${pylintrc}"
+
+    echo > "${pylintrc}"
+    echo "[MESSAGES CONTROL]" >> "${pylintrc}"
+    echo "disable=${disabled_pylint_warning}" >> "${pylintrc}"
+
+    echo "[TYPECHECK]" >> "${pylintrc}"
+    echo "ignored-modules=_cycles" >> "${pylintrc}"
+    echo "ignored-classes=CyclesPreferences" >> "${pylintrc}"
+}
+
+function workaround_quirks() {
+    local version=$1
+
+    if [[ $version =~ ^2.8[0-9]$ ]]; then
+        # There is an issue with CYCLES_PT_sampling_presets.draw_panel_header() requiring two arguments
+        # but only one is delivered.
+        echo "Fixing cycles quirk: \".draw_panel_header(self.layout)\""
+        sed -i 's/.draw_panel_header(self.layout)/.draw_panel_header(self, layout)/' intern/cycles/blender/addon/ui.py
+    fi
+
+    if [[ $version =~ ^2.7[89]$ ]]; then
+        # Grische: I am not sure why, but instead of using the class directly, they are inferering the class
+        # from bpy.types which fake-bpy-module does not provide
+        echo "Fixing cycles quirk: \".bpy.types.CYCLES_MT_[a-z]*_presets\""
+        sed -i 's/bpy.types.\(CYCLES_MT_[a-z]*_presets\)/\1/' intern/cycles/blender/addon/ui.py
+    fi
+
+    if [[ $version =~ ^2.7[89]$ ]]; then
+        # pylint does not respect a `hasattr` in `if hasattr(myclass, field) and myclass.field == test`
+        echo "Ignoring pylint bug: https://github.com/PyCQA/pylint/issues/801"
+        sed -i '/^\s*if hasattr(.*/i # pylint: disable=no-member' intern/cycles/blender/addon/*.py
+    fi
+}
+
+
+function run_pylint_test() {
+    local path=${1}
+    local path_to_pylintrc=${2}
+    local additional_parameters=${3:-}
+
+    local returncode=0
+
+    # shellcheck disable=SC2086
+    PYLINTRC=${path_to_pylintrc} pylint ${additional_parameters} "${path}" || returncode=$?
+
+    return ${returncode}
+}
+
+# Generate a venv, so it does not interfere with local setup
+echo "Creating temporary virtualenv for ${python_bin} at ${temp_venv}"
+${python_bin} -m venv "${temp_venv}"
+# shellcheck source=/dev/null
+source "${temp_venv}"/bin/activate
+
+# install pylint
+pip install --quiet pylint
+
+# Enter blender source
+pushd "${source_dir}"
+
+git_tag="${BLENDER_TAG_NAME[v${version}]}"
+remote_git_ref="$(get_remote_git_ref "${git_tag}")"
+
+echo "Checking out git tag ${git_tag}"
+git fetch --quiet --depth 1 origin "${remote_git_ref}"
+git checkout -f "${remote_git_ref}"
+
+# Generate and print pylintrc
+pylintrcpath="${temp_venv}/pylintrc"
+
+create_pylintrc "${pylintrcpath}"
+echo "Testing with the following pylintrc:"
+cat "${pylintrcpath}"
+echo
```

Comment:
Question: Why echo commands with empty argument are exist after some commands?

Reply:
@nutti it is a nice way to generate an empty line to separate outputs

Reply:
Thanks for your explanation.
I understood.